src/packageManager/AbsolutePath.ts

```diff
@@ -0,0 +1,18 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { isAbsolute, resolve } from "path";
+
+export class AbsolutePath{
+    constructor(public value: string) {
+        if (!isAbsolute(value)) {
+            throw new Error("The path must be absolute");
+        }
+    }
+
+    public static getAbsolutePath(pathToPrepend: string, path: string): AbsolutePath {
```

Comment:
thoughts on making the signature here just look like the signature of path.resolve?

Reply:
Done